security/security/src/main/java/io/helidon/security/Security.java

```diff
@@ -69,7 +69,8 @@
  * @see #builder()
  * @see #create(Config)
  */
-public final class Security {
+// class cannot be final, so CDI can create a proxy for it
```

Comment:
You may discover that you [may need a zero-argument constructor](https://docs.jboss.org/cdi/spec/2.0/cdi-spec.html#unproxyable).  (It can be package-private, and does not have to fully initialize the class.)  Although note that as written this class will not be proxied because it is not annotated with a scope that requires a proxy.  (But you may be programmatically registering it in such a scope elsewhere.)

Reply:
Yes, this is registered by the `SecurityProducer` (part of this PR) in `AppicationScoped`.
The startup failed because CDI was not able to proxy this class. Once I removed the `final` keyword, Security is injected as expected.